src/main/java/at/hannibal2/skyhanni/features/mining/glacitemineshaft/MineshaftWaypoints.kt

```diff
@@ -0,0 +1,89 @@
+package at.hannibal2.skyhanni.features.mining.glacitemineshaft
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.data.IslandType
+import at.hannibal2.skyhanni.data.PartyAPI
+import at.hannibal2.skyhanni.events.IslandChangeEvent
+import at.hannibal2.skyhanni.events.LorenzKeyPressEvent
+import at.hannibal2.skyhanni.events.LorenzRenderWorldEvent
+import at.hannibal2.skyhanni.events.LorenzWorldChangeEvent
+import at.hannibal2.skyhanni.test.GriffinUtils.drawWaypointFilled
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.LocationUtils
+import at.hannibal2.skyhanni.utils.LocationUtils.distanceToPlayer
+import at.hannibal2.skyhanni.utils.LorenzUtils.isInIsland
+import at.hannibal2.skyhanni.utils.RenderUtils.drawDynamicText
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import net.minecraft.client.Minecraft
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.milliseconds
+
+object MineshaftWaypoints {
+    private val config get() = SkyHanniMod.feature.mining.mineshaftWaypoints
+
+    @Suppress("ConstPropertyName")
+    private const val blocksForward: Int = 7
+
+    val waypoints: MutableList<Waypoint> = mutableListOf()
+    private var timeLastShared = SimpleTimeMark.farPast()
+
+    @Suppress("UNUSED_PARAMETER")
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        waypoints.clear()
+    }
+
+    @SubscribeEvent
+    fun onIslandChange(event: IslandChangeEvent) {
+        if (event.newIsland != IslandType.MINESHAFT) return
+
+        val playerLocation = LocationUtils.playerLocation().round(0).add(y = -1)
```

Comment:
This can someones be off by a block or two leading to the waypoint to be off slightly but there is no way to avoid this really.